core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -727,6 +727,10 @@ private NullAs negate(NullAs nullAs) {
             return NullAs.TRUE;
           case TRUE:
             return NullAs.FALSE;
+          case IS_NULL:
+            return NullAs.IS_NOT_NULL;
+          case IS_NOT_NULL:
+            return NullAs.IS_NULL;
           default:
             return nullAs;
```

Comment:
Before this change, the generated code for conditions like `name like 'ss' and name not like 'xx'` is (inp1_ == null || inp1_ != null ? (Boolean) null : Boolean.TRUE)` which is wrong because it always return null. After this change it becomes
`(inp1_ == null ? (Boolean) null : Boolean.TRUE) : Boolean.FALSE`.